### PR TITLE
[glTF] Update binary (.GLB) parsing for glTF 2.0.

### PR DESCRIPTION
**Pull request by donmccurdy**
This implements the new [GLB File Format Specification](https://github.com/KhronosGroup/glTF/blob/2.0/specification/2.0/GLB_FORMAT.md).

Note that this PR is _not_ backward-compatible with glTF 1.0 models. I'm assuming that's the plan for having GLTF2Loader vs GLTFLoader and overall code health, but I think this is the first change that breaks compatibility, so just FYI.

### CODE REVIEW

**Comment by takahirox** [comment]
With this change, `GLTF2Loader` is no longer able to load the glTF 1.0 binary models under examples/models/gltf?

**Comment by donmccurdy** [comment]
Correct. I don't see a case for having two branched loaders, and also branched code within this one.

Should I remove the other glTF-Binary models from the demo page? Update the demo to switch between GLTFLoader and GLTF2Loader as needed?

**Comment by mrdoob** [comment]
I would even suggest to remove `GLTFLoader` altogether 😇

**Comment by takahirox** [comment]
I think we have four options.

1. Remove glTF 1.0 binary models from `examples/models/gltf`
2. Switch between `GLTFLoader` and `GLTF2Loader` as needed
3. Make `examples/webgl_loader_gltf2.html`
4. Remove `GLTFLoader` and 1.0 binary models now

I prefer 3.

glTF 2.0 isn't officially released yet, so it'd be a bit early to remove glTF 1.0 binary models from example.
But I don't wanna be bothered to switch `GLTF(2)Loader` as needed.
We can marge/replace `webgl_loader_gltf(2).html` as we marge/replace `GLTF(2)Loader`.

If we decide Three.js doesn't support glTF 1.0 anymore even 2.0 is released,
4. is also a good option.

**Comment by mrdoob** [comment]
I vote for 3 too.

**Comment by donmccurdy** [comment]
Sounds good to me. Opened separately as #10972.

File: examples/js/loaders/GLTF2Loader.js [hunk 1]
```diff
@@ -1070,8 +1077,9 @@ THREE.GLTF2Loader = ( function () {
 						if ( source.bufferView !== undefined ) {
 
 							var bufferView = dependencies.bufferViews[ source.bufferView ];
-							var stringData = convertUint8ArrayToString( new Uint8Array( bufferView ) );
-							sourceUri = 'data:' + source.mimeType + ';base64,' + btoa( stringData );
+							var blob = new Blob( [ bufferView ], { type: source.mimeType } );
+							var urlCreator = window.URL || window.webkitURL;
+							sourceUri = urlCreator.createObjectURL( blob );
```
Comment (mrdoob) [inline examples/js/loaders/GLTF2Loader.js]: Do we need to `URL.revokeObjectURL( blob )` also?

Comment (donmccurdy) [inline examples/js/loaders/GLTF2Loader.js]: Good call, thanks!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**